src/app/teacher/manage-tags-dialog/manage-tags-dialog.component.ts

```diff
@@ -70,4 +81,28 @@ export class ManageTagsDialogComponent implements OnInit {
       panelClass: 'dialog-md'
     });
   }
+
+  protected delete(tag: Tag): void {
+    this.teacherService.getRuns().subscribe((runs) => {
+      if (confirm(this.getDeleteMessage(runs, tag))) {
+        this.projectTagService.deleteTag(tag);
+      }
+    });
+  }
+
+  private getDeleteMessage(runs: Run[], tag: Tag): string {
+    const numberOfProjectsWithTag = this.getNumberOfProjectsWithTag(runs, tag);
+    const numberOfProjectsMessage =
+      numberOfProjectsWithTag === 1
+        ? $localize`There is ${numberOfProjectsWithTag} unit with this tag.`
+        : $localize`There are ${numberOfProjectsWithTag} units with this tag.`;
+    return $localize`Are you sure you want to delete this tag? ` + numberOfProjectsMessage;
+  }
+
+  private getNumberOfProjectsWithTag(runs: Run[], tag: Tag) {
+    const projectsWithTag = runs.filter((run: Run) =>
+      run.project.tags.some((projectTag: Tag) => projectTag.id === tag.id)
+    );
+    return projectsWithTag.length;
```

Comment:
I don't think we need a variable here. We may not even need this function since it's only called in one place?
```suggestion
    return runs.filter((run: Run) =>
      run.project.tags.some((projectTag: Tag) => projectTag.id === tag.id)
    ).length;
```